allennlp/data/dataset_readers/wikitables/world.py

```diff
@@ -0,0 +1,143 @@
+        follow NLTK's variable naming conventions.
+        """
+        expressions = []
+        for sempre_form in sempre_forms:
+            parsed_lisp = pyparsing.OneOrMore(pyparsing.nestedExpr()).parseString(sempre_form).asList()
+            translated_string = self._process_nested_expression(parsed_lisp)
+            expression = self._logic_parser.parse(translated_string, signature=self.type_signature)
+            expressions.append(expression)
+        return expressions
+
+    def _process_nested_expression(self, nested_expression):
+        """
+        ``nested_expression`` is the result of parsing a Lambda-DCS expression in Lisp format. We process it
+        recursively and return a string in the format that NLTK's ``LogicParser`` would understand.
+        """
+        expression_is_list = isinstance(nested_expression, list)
+        expression_size = len(nested_expression)
+        if expression_is_list and expression_size == 1 and isinstance(nested_expression[0], list):
+            return self._process_nested_expression(nested_expression[0])
+        elements_are_leaves = [isinstance(element, str) for element in nested_expression]
+        if all(elements_are_leaves):
+            mapped_names = [self._map_name(name) for name in nested_expression]
+        else:
+            mapped_names = []
+            for element, is_leaf in zip(nested_expression, elements_are_leaves):
+                if is_leaf:
+                    mapped_names.append(self._map_name(element))
+                else:
+                    mapped_names.append(self._process_nested_expression(element))
+        if mapped_names[0] == "\\":
+            # This means the predicate is lambda. NLTK wants the variable name to not be within parantheses.
+            arguments = [mapped_names[1]] + ["(%s)" % name for name in mapped_names[2:]]
+        else:
+            arguments = ["(%s)" % name for name in mapped_names[1:]]
+        return "(%s %s)" % (mapped_names[0], " ".join(arguments))
+
+    def _map_name(self, sempre_name: str) -> str:
+        """
+        Takes the name of a predicate or a constant as used by Sempre, maps it to a unique string such that
+        NLTK processes it appropriately. This is needed because NLTK has a naming convention for variables:
+            Function variables: Single upper case letter optionally foolowed by digits
+            Individual variables: Single lower case letter (except e for events) optionally followed by digits
+            Constants: Everything else
+
+        Parameters
+        ----------
+        sempre_name : str
+            Token from Sempre's logical form.
+        """
+        if sempre_name not in self.name_mapping:
+            if sempre_name.startswith("fb:row.row"):
+                # Column name
+                translated_name = "C%d" % self._column_counter
+                self._column_counter += 1
+                self.type_signature[translated_name] = COLUMN_TYPE
```

Comment:
It sure looks like you could use that three-argument function I mentioned earlier here, too.